ospfd/ospf_vty.c

```diff
@@ -9300,13 +9300,12 @@ DEFUN (ospf_default_information_originate,
 	if (argv_find(argv, argc, "WORD", &idx))
 		rtmap = argv[idx]->arg;
 
-	/* To check ,if user is providing same route map */
-	if ((rtmap == ROUTEMAP_NAME(red)) ||
-	    (rtmap && ROUTEMAP_NAME(red)
-	    && (strcmp(rtmap, ROUTEMAP_NAME(red)) == 0)))
+	/* To check if user is providing same route map */
+	if (rtmap && ROUTEMAP_NAME(red)
```

Comment:
Removing this check "rtmap == ROUTEMAP_NAME(red)" will lead to below issue:
if user execute "default-information originate" command multiple times without the route-map option, the default LSAs will be re-originated.